app/web_form_handlers.py

```diff
@@ -0,0 +1,184 @@
+import aiohttp_jinja2
+import re
+
+from aiohttp.client_exceptions import (ClientResponseError)
+from aiohttp.web import HTTPFound, RouteTableDef
+from structlog import get_logger
+
+from . import (WEBFORM_MISSING_COUNTRY_MSG,
+               WEBFORM_MISSING_QUERY_MSG,
+               WEBFORM_MISSING_DESCRIPTION_MSG,
+               WEBFORM_MISSING_NAME_MSG,
+               WEBFORM_MISSING_EMAIL_EMPTY_MSG,
+               WEBFORM_MISSING_EMAIL_INVALID_MSG,
+               WEBFORM_MISSING_COUNTRY_MSG_CY,
+               WEBFORM_MISSING_QUERY_MSG_CY,
+               WEBFORM_MISSING_DESCRIPTION_MSG_CY,
+               WEBFORM_MISSING_NAME_MSG_CY,
+               WEBFORM_MISSING_EMAIL_EMPTY_MSG_CY,
+               WEBFORM_MISSING_EMAIL_INVALID_MSG_CY
+               )
+from .flash import flash
+from .utils import View, RHService
+
+logger = get_logger('respondent-home')
+web_form_routes = RouteTableDef()
+
+
+class WebForm(View):
+    email_validation_pattern = re.compile(
+        r'(^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$)'
```

Comment:
as discussed, consider a simpler Regex . Just not sure whether the 99.9% case is good enough ?

Reply:
Simplified - ^[^@\s]+@[^@\s]+\.[^@\s]+$
New pattern only validates for a single '@', with non-whitespace before, and non-whitespace and at least one '.' after